### PR TITLE
[lldb] Add Foundation._NSSwiftTimeZone support

### PR DESCRIPTION
**Pull request by kastiglione**
Cherry pick the two missing commits from #7074.

### CODE REVIEW

**Comment by kastiglione** [comment]
@swift-ci test

File: lldb/source/Plugins/Language/ObjC/Cocoa.cpp [hunk 1]
```diff
@@ -131,6 +131,20 @@ bool lldb_private::formatters::NSTimeZoneSummaryProvider(
       stream.Printf("%s", summary_stream.GetData());
       return true;
     }
+  } else if (class_name == "_NSSwiftTimeZone") {
+    llvm::ArrayRef<ConstString> identifier_path = {
+        ConstString("timeZone"),
+        ConstString("_timeZone"),
+        ConstString("some"),
+        ConstString("identifier"),
```
Comment (adrian-prantl) [inline lldb/source/Plugins/Language/ObjC/Cocoa.cpp]: This almost looks like some leftover debugging code?

Comment (kastiglione) [reply lldb/source/Plugins/Language/ObjC/Cocoa.cpp]: It's not debugging code. That's the variable path that contains the time zone's name (effectively `timeZone._timeZone.identifier`).

**Comment by kastiglione** [comment]
@swift-ci test

File: lldb/source/Plugins/Language/ObjC/Cocoa.cpp (hunk 1 above)
Comment (adrian-prantl) [reply lldb/source/Plugins/Language/ObjC/Cocoa.cpp]: Oh right `.some` is the enumerator for Optional!
"some identifier"  just sounded like a placeholder :-)

**Review by adrian-prantl — APPROVED** [review]

**Comment by adrian-prantl** [comment]
Can you cherry-pick this to 5.9 and 5.10?

**Comment by kastiglione** [comment]
@swift-ci test windows

**Comment by kastiglione** [comment]
@swift-ci test windows

**Comment by adrian-prantl** [comment]
@swift-ci test windows

**Pull request merged**